Docs: Fixes and improvements in Installation section, part 1

andy-maier opened the pull request:
This PR addresses a first part of changes to the "Installation" chapter of the documentation, and addresses some items from issue #1067 in that area (see there for status).
For details, see the commit message.
Ready for review and merge. Review is best done via the built HTML docs.

KSchopmeyer commented on docs/intro.rst:
```diff
@@ -201,25 +227,21 @@ Installing to OS-X
-Note that Unix-like environments on Windows bring their own Python, so double
-check that the active Python environment is the one you want to install to.
+This section describes the installation of pywbem into a native Windows
+environment (i.e. without using a UNIX-like environment such as CygWin, etc.):
```
Add ref to section for installing under cygwin

andy-maier replied:
DONE

KSchopmeyer commented on docs/intro.rst:
```diff
@@ -239,14 +261,38 @@ environment. You can detect that bit size as follows:
+
+  If the Swig compilation during installation of M2Crypto fails, there may be
+  components of the UNIX-like environment missing (e.g. OpenSSL development,
+  Python development, Swig, gcc).
```
Add the following: At least under cygwin, these may be installed using the cygwin installer (openssl-devel, python2-devel or python3-devel), swig).

andy-maier replied:
Two comments:
* python3-devel will never be needed, because M2Crypto is only used on Python 2.
* The gcc package on cygwin is called `gcc-g++`.
I have added the information to the package prereq table, and the paragraph you commented on now references that section.
DONE.

KSchopmeyer commented on docs/intro.rst:
```diff
@@ -239,14 +261,38 @@ environment. You can detect that bit size as follows:
+
+  .. code-block:: bash
+
+      > pip install pywbem
```
Add something to reference how pywbem is installed from github.  It is discussed in following section.

andy-maier replied:
That would apply to all target OSs. Plus, the section about using a virtual python env would also apply to all of them. Atthe end, there are three dimensions (target OS, python env, pywbem version) and the cross product of them can actually be used. I don't think we should start listing that cross product. So far, the description starts simple (with just the list of target OSs), and then the reader learns that there are two more dimensions. Each new dimension is introduced by explaining that it applies to all of the variations before.

If we stick with this approach, I don't think we should add the github information to the section about the UNIX-like env.

KSchopmeyer commented on docs/intro.rst:
```diff
@@ -418,7 +466,15 @@ in this table; you can find them in section
-| Windows                  | None               |                      |       |
+| Windows (native)         | None               |                      |       |
++--------------------------+--------------------+----------------------+-------+
+| Windows (UNIX-like env)  | OpenSSL developm.  |                      | py2   |
```
The name is openssl-devel

andy-maier replied:
DONE. I updated all packages and rescoped that group of table rows to only CygWin.

KSchopmeyer commented on docs/intro.rst:
```diff
@@ -640,52 +679,61 @@ Configuration variables
-1. OpenWBEM server - includes support for an authentication extension
-(OWLocal password-less local authorization) that is part of the OpenWBEM server.
+   - Pywbem supports the upper-case variant ``EMBEDDEDOBJECT`` of the respective
+     CIM-XML attribute that is specific to OpenPegasus, in addition to the
```
change to "some releases of OpenPegasus"

andy-maier replied:
DONE.

KSchopmeyer commented:
NEW COMMENTS: Tried to install pywbem directly into windows on another vm.

This uses the pip install M2Crypto...

With python this fails on the released pywbem (known issues) but works with the checkout from github.


1. The pip install of M2cryptoWin64 works fine.

2. The pip install passes and wbemcli is available as is wbemcli.bat.  Both start.
(pywbem27) C:\Users\kschopmeyer\python\pywbemtest>wbemcli http://sheldon

Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (
AMD64)]

Wbemcli interactive shell
Connection: http://sheldon, no creds, cacerts=sys-default, verifycert=on,
    default-namespace=root/cimv2 stats=on, log=off,
Press Ctrl-D to exit
Type h() for help
>>> cls = gc('cim_managedelement')
>>> print cls.tomof()
   [Abstract ( true ),
    Version ( "2.19.0" ),
. . .

};

>>> help
Type help() for interactive help, or help(object) for help about object.
>>> quit()

(pywbem27) C:\Users\kschopmeyer\python\pywbemtest>wbemcli.bat http://sheldon

Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (
AMD64)]

Wbemcli interactive shell
Connection: http://sheldon, no creds, cacerts=sys-default, verifycert=on,
    default-namespace=root/cimv2 stats=on, log=off,
Press Ctrl-D to exit
Type h() for help
>>>

andy-maier commented:
Karl, on your comment "NEW COMMENTS: Tried to install pywbem directly into windows on another vm.":

The installation section documents for native Windows (assuming 64-bit Python 2.7):
```
pip install M2CryptoWin64
pip install pywbem
```
I assume this is what you did.
The point is that when you *currently* install pywbem from pypi, it will use pywbem 0.11, which indeed fails because it tries to install `M2Crypto`.

However, installing pywbem 0.12 from pypi, once it is released, will work.
You can "simulate" that now by installing the work directory of your pywbem git repo clone:
```
pip install .
```

andy-maier commented:
Updated the PR with a commit that addresses the comments, as stated above.

Karl: I have set "review needed" again on this PR, with the following TODOs for you:
* Please verify whether the changes in the last commit address your comments.
* Please check my response on your comment about installing from github.
* Please verify that `pip install .` works for you, see my previous comment.

KSchopmeyer commented:
I am satisfied with changes.  Will run tests again.

Actually get the issue when I install pip install.  See below

``
... Do fresh clone
C:\Users\kschopmeyer\python\pywbem>git status
On branch andy/#1067-installation
Your branch is up-to-date with 'origin/andy/#1067-installation'.
nothing to commit, working directory clean
C:\Users\kschopmeyer\python\pywbem>pip install M2CryptoWin64
Collecting M2CryptoWin64
Installing collected packages: M2CryptoWin64
Successfully installed M2CryptoWin64-0.21.1.post3

C:\Users\kschopmeyer\python\pywbem>pip install .
Processing c:\users\kschopmeyer\python\pywbem
Requirement already satisfied: pbr>=1.10.0 in c:\python27\lib\site-packages (fro
m pywbem==0.11.1.dev372)
Requirement already satisfied: six>=1.10.0 in c:\python27\lib\site-packages\six-
1.10.0-py2.7.egg (from pywbem==0.11.1.dev372)
Requirement already satisfied: ply>=3.10 in c:\python27\lib\site-packages (from
pywbem==0.11.1.dev372)
Requirement already satisfied: PyYAML>=3.12 in c:\python27\lib\site-packages (fr
om pywbem==0.11.1.dev372)
Requirement already satisfied: mock>=2.0.0 in c:\python27\lib\site-packages (fro
m pywbem==0.11.1.dev372)
Requirement already satisfied: funcsigs>=1; python_version < "3.3" in c:\python2
7\lib\site-packages (from mock>=2.0.0->pywbem==0.11.1.dev372)
Installing collected packages: pywbem
  Running setup.py install for pywbem ... done
Successfully installed pywbem-0.11.1.dev372

NOTE: Install 0.11.1

andy-maier commented:
That's correct. The version as determined by pbr is the latest version tag in the checked-out branch, and it increases the update version by 1 and adds the `.devNNN` suffix.

KSchopmeyer commented:
Lets commit this
